src/core/evaluator.js

```diff
@@ -2656,20 +2654,26 @@ class PartialEvaluator {
       textContentItem.str.length = 0;
     }
 
-    function enqueueChunk() {
+    function enqueueChunk(batch = false) {
       const length = textContent.items.length;
-      if (length > 0) {
-        sink.enqueue(textContent, length);
-        textContent.items = [];
-        textContent.styles = Object.create(null);
+      if (length === 0) {
+        return;
+      }
+      if (batch && length < 10) {
+        // Avoid sending individual, or very few, text chunks to reduce
+        // `postMessage` overhead (see issue 13962).
+        return;
```

Comment:
I might be missing something here, but if the last batch to be processed has a size < 10, doesn't that mean we'll never enqueue it at all and basically drop it?

Reply:
Note how at the *end* of the "GetTextContent"-parsing we'll do one final sending of any remaining chunks:
https://github.com/mozilla/pdf.js/blob/168aa5efb12e3c0f6a0a1713cde3717b337c3484/src/core/evaluator.js#L3085-L3101

Furthermore, I very deliberately implemented the batching to be *opt-in* precisely to avoid the problem you're asking about here; so unless I've somehow messed up here we should be good.

---

With the exception of some movement in a couple of "enhance" text-tests, which aren't really my fault (note footnote [2] above), this patch also passes all regular "text" tests locally.

Reply:
Yes, the finalization step is the part I missed here, and `enqueueChunk` is indeed called without the `batch` parameter, so we should indeed be good here.